src/cpp/client/secure_credentials.cc

```diff
@@ -280,9 +280,9 @@ std::shared_ptr<ChannelCredentials> LocalCredentials(
 
 // Builds TLS Credentials given TLS options.
 std::shared_ptr<ChannelCredentials> TlsCredentials(
-    const TlsCredentialsOptions& options) {
+    const std::shared_ptr<TlsCredentialsOptions>& options) {
```

Comment:
Why do we need a `shared_ptr`? Credentials and options are usually 1:1 mapping

Reply:
Reverted back to constant reference in the next commit.